src/mixed_arena.h

```diff
@@ -277,31 +277,101 @@ class ArenaVectorBase {
   // iteration
 
   struct Iterator {
+    using iterator_category = std::random_access_iterator_tag;
+    using value_type = T;
+    using difference_type = std::ptrdiff_t;
+    using pointer = T*;
+    using reference = T&;
+
     const SubType* parent;
     size_t index;
 
+    Iterator() : parent(nullptr), index(0) {}
     Iterator(const SubType* parent, size_t index) : parent(parent), index(index) {}
 
+    bool operator==(const Iterator& other) const {
+      return index == other.index && parent == other.parent;
+    }
+
     bool operator!=(const Iterator& other) const {
-      return index != other.index || parent != other.parent;
+      return !(*this == other);
+    }
+
+    bool operator<(const Iterator& other) const {
+      return (parent->data + index) < (other.parent->data + other.index);
```

Comment:
I think it's enough to compare the indexes here - as the parent must be identical? (perhaps we should assert on that?)

Reply:
Since `a < b` must be equal to `b - a > 0`  [*1], I agree that it is better to order only among iterators with the same parent.
509dd76cb8c923aa5da717d1a5a81c31f9748524
*1: https://en.cppreference.com/w/cpp/named_req/RandomAccessIterator